src/utils/useGetGasParams.ts

```diff
@@ -46,17 +32,16 @@ export const useGetGasParams = (
         .then((m: Message) => {
           const gasFeeCap = m.gasFeeCap.toFixed(0, BigNumber.ROUND_CEIL)
           const gasPremium = m.gasPremium.toFixed(0, BigNumber.ROUND_CEIL)
-          const newGasParams = {
+          setGasParams({
             gasFeeCap: new FilecoinNumber(gasFeeCap, 'attofil'),
             gasPremium: new FilecoinNumber(gasPremium, 'attofil'),
             gasLimit: new FilecoinNumber(m.gasLimit, 'attofil')
-          }
-          if (didUpdate(newGasParams)) setGasParams(newGasParams)
+          })
         })
         .catch((e: Error) => setError(e))
         .finally(() => setLoading(false))
     }
-  }, [provider, message, maxFee, didUpdate])
+  }, [provider, message, maxFee])
```

Comment:
@Schwartz10 This had to be reverted. Basically what it creates is a condition where we first set `gasParams` to `null` and then we never update it after because our cached values are still the same. This is actually very easy to reproduce and creates quite a problem.

The issue we were trying to solve here as I remember is to keep the values when the instance of provider changes. I'm not sure if we should worry about it too much, as it sounded like an edge case? If it does happen, we just reload the gas params once more, it doesn't seem like a biggie.